tokio-util/src/udp/frame.rs

```diff
@@ -34,8 +37,8 @@ use std::{io, mem::MaybeUninit};
 #[must_use = "sinks do nothing unless polled"]
 #[cfg_attr(docsrs, doc(all(feature = "codec", feature = "udp")))]
 #[derive(Debug)]
-pub struct UdpFramed<C> {
-    socket: UdpSocket,
+pub struct UdpFramed<T, C> {
+    socket: T,
```

Comment:
This is a breaking change, and would require a new tokio-util series.

Reply:
Yeah, although I don't see any other way to enable this:

```
    let a = Arc::new(UdpSocket::bind("127.0.0.1:0").await?);
    let b = Arc::clone(&a);

    let mut recv = UdpFramed::new(a, ByteCodec);
    let mut send = UdpFramed::new(b, ByteCodec);
```

I think it's a fairly common scenario that you'll have a `UdpSocket` in an `Arc` in order to share between different tasks? You may want to have one task doing the sending and one doing the recv'ing for example.

Reply:
I think it would be possible to add it with a default as in `UdpFramed<C, T = UdpSocket>`.

Reply:
Sure, I'll experiment with that and report back. Probably best to switch the type param order as you have there also.

Reply:
That seemed to work. I was also able to add back `get_mut` by writing an impl block specifically for `UdpFramed<C>` using the default param.